File: job_message_writer/backend/app/schemas/message.py
# File: backend/app/schemas/message.py
from pydantic import BaseModel
from typing import Dict, Any, Optional, List

class ProfileClassification(BaseModel):
    profile_type: Optional[str] = None
    primary_languages: Optional[List[str]] = None
    frameworks: Optional[List[str]] = None
    years_experience: Optional[str] = None
    seniority: Optional[str] = None
    industry_focus: Optional[str] = None

class MessageRequest(BaseModel):
    resume_id: Optional[int] = None  # Optional - if not provided, use active profile
    job_description: str
    message_type: str  # "linkedin", "inmail", "email", "ycombinator"
    recruiter_name: Optional[str] = None  # Optional recruiter name

class MessageResponse(BaseModel):
    message: str
    company_info: Dict[str, Any]
    resume_info: Optional[Dict[str, Any]] = None
    profile_classification: Optional[ProfileClassification] = None
    resume_id: int  # Which resume was used
    resume_title: str  # The title of the resume profile

File: job_message_writer/frontend-old/src/api/message.ts
import apiClient from './client';
import { CompanyInfo } from './job';

// Types
export interface ProfileClassification {
  profile_type: string;
  primary_languages: string[];
  frameworks: string[];
  years_experience: string;
  seniority: string;
  industry_focus: string;
}

export interface MessageResponse {
  message: string;
  company_info: CompanyInfo;
  resume_info: Record<string, any>;
  profile_classification: ProfileClassification;
  resume_id: number;
  resume_title: string;
}

export type MessageType = 
  | 'linkedin_message' 
  | 'linkedin_connection' 
  | 'linkedin_inmail' 
  | 'email_short' 
  | 'email_detailed'
  | 'ycombinator';
// Generate a message
export const generateMessage = async (
  jobDescription: string,
  messageType: MessageType,
  resumeId?: number,
  recruiterName?: string
) => {
  const response = await apiClient.post<MessageResponse>('/messages/generate', {
    job_description: jobDescription,
    message_type: messageType,
    resume_id: resumeId,
    recruiter_name: recruiterName
  });
  
  return response.data;
};